### PR TITLE
Content mapping with FS 896

### PR DESCRIPTION
**Pull request by ramsharma-prog**
A very simple update on notification service to map the data with Magic_link template from ticket FS-896. Tests contents to be updated when template ID has been updated to magic_link_template_ID on CF. Temporarily tests are running on test_template_id & work fine

### CODE REVIEW

File: app/notification/models/notification.py [hunk 1]
```diff
@@ -66,3 +69,38 @@ def send_reminder(contact_info, content):
...
+                    ),
+                },
+            )
+            return response
+
+        except (TypeError, KeyError, AttributeError):
+            example_data = json.dumps(get_example_data(), indent=2)
+            return (
+                "Bad request, please check the contents of the notification"
+                f" data.\n\n Example data: {example_data}"
```
Comment (gidsg) [inline app/notification/models/notification.py:105]: Why are we using example data here?

Comment (ramsharma-prog) [reply app/notification/models/notification.py:105]: The contents have to be with certain keys & a type so if there was some sort of mistake from other services while posting data to the notification service, returning an example data would be helpful?

**Review by harryyo — APPROVED** [review]

File: app/notification/magic_link/process_contents.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+    Returns: notification data or updated notification data
+    to map out with expected contents.
+    """
+
+    @staticmethod
+    def fund_name(data):
+        try:
+            if "fund_name" not in data["content"]:
+                data["content"].update(fund_name="Funds")
```
[the remark this answers is not in the document]
Comment (ramsharma-prog) [reply app/notification/magic_link/process_contents.py]: Thanks, Harry! That is some real professional code 😄 
Updated!

**Review by harryyo — APPROVED** [review]

File: app/notification/template_types.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    elif data.get("type") == "NOTIFICATION":
+        return f"Currently {data.get('type')} service is not available."
+    elif data.get("type") == "REMINDER":
+        return f"Currently {data.get('type')} service is not available."
+    elif data.get("type") == "AWARD":
+        return f"Currently {data.get('type')} service is not available."
+    else:
+        raise NotificationError(
+            message=(
+                "Incorrect type, please check the key 'type' & other keys,"
```
Comment (gidsg) [inline app/notification/template_types.py:17]: Could we use pattern matching instead e.g https://peps.python.org/pep-0636/

File: app/notification/routes.py [hunk 4]
```diff
@@ -18,7 +20,7 @@
 )
 
 
-@notification_bp.route("/", methods=["POST", "GET"])
+@notification_bp.route("", methods=["POST", "GET"])
```
Comment (gidsg) [inline app/notification/routes.py:23]: Why does this accept GET when it is expecting a POST?

Comment (ramsharma-prog) [reply app/notification/routes.py:23]: For some reason, it wasn't working without the GET method but now I have tried without it, it works so updated!

File: README.md [hunk 5]
```diff
@@ -43,11 +43,13 @@ Enter the virtual environment as described above, then:
 ## How to post data.
 To post data, use /send endpoint with POST method. Data must be in the following format.
 
- {
-    "type": "TEST_MAGIC_LINK",
-    "to": "test_recipient@email.com",
-    "content": "MAGIC LINK GOES HERE"
-}
+   {
```
Comment (endamccormack) [inline README.md:46]: are you sure the tab padding is right in this json? looks odd

File: README.md [hunk 6]
```diff
@@ -43,11 +43,13 @@ Enter the virtual environment as described above, then:
...
+    "type": "MAGIC_LINK",
+    "to": "ramandeep.sharma@communities.gov.uk",
+    "content": {"magic_link_url":"https://www.google.co.uk/",
```
Comment (endamccormack) [inline README.md:49]: i'd hit a new line after the opening of the curley and before the end of the block

File: tests/test_route.py [hunk 7]
```diff
@@ -50,13 +56,16 @@ def test_notification_route_response_unexpected_data(flask_test_client):
     """
     GIVEN: our service running on flask test client.
     WHEN: we post unexpected data to the endpoint "/send".
-    THEN: we expect a successful response 200.
+    THEN: we expect a unsuccessful response 400.
     """
 
     unexpected_data = {
-        "tpe": "MAGIC_LINK",
+        "te": "MAGIC_LINK",
```
Comment (endamccormack) [inline tests/test_route.py:63]: why did you change the field name?

File: app/notification/template_types.py [hunk 8]
```diff
@@ -2,21 +2,25 @@
...
+            return notification_class.send_magic_link(data)
+
+        case "NOTIFICATION":
+            return f"Currently {data.get('type')} service is not available."
+
+        case "REMINDER":
+            return f"Currently {data.get('type')} service is not available."
+
+        case "AWARD":
+            return f"Currently {data.get('type')} service is not available."
```
Comment (gidsg) [inline app/notification/template_types.py:16]: You could use an Or pattern here and simplify this https://peps.python.org/pep-0636/#or-patterns

File: README.md (hunk 5 above)
Comment (ramsharma-prog) [reply README.md:46]: This is old JSON 🤔

File: README.md (hunk 6 above)
Comment (ramsharma-prog) [reply README.md:49]: Thank you. Updated 😄

File: app/notification/template_types.py (hunk 8 above)
Comment (ramsharma-prog) [reply app/notification/template_types.py:16]: Thank you for pointing that out, not sure why didn't I think of it. Updated 😄

File: tests/test_route.py (hunk 7 above)
Comment (ramsharma-prog) [reply tests/test_route.py:63]: It is testing the incorrect key so made it a bit obvious that the key is incorrect 😄

File: README.md (hunk 5 above)
Comment (ramsharma-prog) [reply README.md:46]: It was incorrect formatting. Thank you for pointing that out. Updated!
